src/app/text-marking/text-marking.ts

```diff
@@ -101,6 +114,138 @@ export function markText(
             }
         }
     }
+
+    isLastCall(textMarkings.length, typoElements.length);
+}
+
+/**
+ * Updates the state of text markings and applies animations to the HTML element based on changes detected
+ * after invoking the `markText` function.
+ *
+ * @param {HTMLElement} newNode - The newly created or modified HTML element.
+ * @param {string} newTextContent - The text content of the HTML element.
+ *
+ * This function adds the new text content to the `highlightedMarkingWords` array.
+ * It then checks the previous state (`lastHighlightedMarkingWords`) and determines
+ * whether to apply animations to the nodes based on the changes detected.
+ * The logic considers scenarios where nodes are added, removed, or modified.
+ */
+function makeTextMarkings(newNode: HTMLElement, newTextContent: string) {
```

Comment:
This method is now much more readable! Btw I feel like the terminology can be improved a bit, i.e. can we rename this from `makeTextMarkings` to `applyMarkingFadein`, or something else that is a bit more descriptive.